src/lib/api/coaching-sessions.ts

```diff
@@ -123,6 +129,50 @@ export const CoachingSessionApi = {
   ): Promise<CoachingSession> => {
     throw new Error("Delete nested operation not implemented");
   },
+
+  /**
+   * Fetches coaching sessions for a specific user with optional related data.
+   *
+   * This uses the enhanced /users/{user_id}/coaching_sessions endpoint that
+   * supports batch loading of related resources to avoid N+1 queries.
+   *
+   * @param userId The ID of the user (coach or coachee)
+   * @param fromDate Start date for filtering sessions
+   * @param toDate End date for filtering sessions
+   * @param include Optional array of related resources to include
+   * @param sortBy Optional field to sort by
+   * @param sortOrder Optional sort order
+   * @returns Promise resolving to array of EnrichedCoachingSession objects
+   */
+  listForUser: async (
```

Comment:
Potentially naive thought:

Could we use the pattern that we use for [createNested](https://github.com/refactor-group/refactor-platform-fe/blob/main/src/lib/api/entity-api.ts#L453) here?

meaning add a `listNested` function to the base `EntityApi` and use that here?

Reply:
It's a good suggestion and I think so, let me think about this a bit more.